### ISSUE TITLE
renames: ensure that rename destinations don't overlap across rename entries

### ISSUE THREAD

**Issue by listx** [kubernetes-sigs/k8s-container-image-promoter#43]
A user could mistakenly do this in the manifest:

```
renames:
- ["gcr.io/foo/a", "gcr.io/bar/subdir/a"]
- ["gcr.io/foo/b", "gcr.io/bar/subdir/a"]
```

where both `a` and `b` will get renamed to `subdir/a`. We need to catch this as an error during the Manifest parsing phase.

**Comment by listx** [comment]
This is now tied to #64, because that will make the promoter act on a set of manifests in a single invocation. This means reconciling all of the renames as well.

**Comment by listx** [comment]
This was fixed as part of #64.

/close
